### PR TITLE
Update glossary

### LINKED ISSUE #2187: Update glossary with latest QEM techniques
The [glossary page](https://github.com/unitaryfund/mitiq/blob/master/docs/source/guide/glossary.md) can be a great resource for people breaking in to QEM, but it does not include all of the latest QEM techniques that are included in Mitiq. Resolving this issue means updating it to include all of Mitiq's currently supported techniques (QSE, shadows, and pauli twirling are missing[^1]). Any terms that should be included in the glossary relating to the missing techniques should also be added.

This is a follow on from creating the glossary in https://github.com/unitaryfund/mitiq/issues/1582.

[^1]: Make sure there aren't any others!

### PR DESCRIPTION
**Pull request by purva-thakre**
## Description

Fixes #2187 

---

### License

- [ ] I license this contribution under the terms of the GNU GPL, version 3 and grant Unitary Fund the right to provide additional permissions as described in section 7 of the GNU GPL, version 3.

Before opening the PR, please ensure you have completed the following where appropriate.

- [ ] I added unit tests for new code.
- [ ] I used [type hints](https://www.python.org/dev/peps/pep-0484/) in function signatures.
- [ ] I used [Google-style](https://google.github.io/styleguide/pyguide.html#383-functions-and-methods) docstrings for functions.
- [ ] I [updated the documentation](../blob/main/docs/CONTRIBUTING_DOCS.md) where relevant.
- [ ] Added myself / the copyright holder to the AUTHORS file

### CODE REVIEW

File: docs/source/guide/glossary.md [hunk 1]
```diff
@@ -52,6 +59,9 @@ between the qubits and the environment, mitigating the effects of noise.
 [Probabilistic Error Cancellation (PEC)](pec.md)
 : Ideal operations are represented as quasi-probability distributions over noisy implementable operations, and unbiased estimates of expectation values are obtained by averaging over circuits sampled according to this representation.
 
+[Quantum Subspace Expansion](qse.md)
+: The error-mitigated expectation value of some observable is estimated by searching a subspace of an output quantum state with the lowest error rate without utilizing syndrome measurements often required by quantum error correcting schemes.
```
Comment (purva-thakre) [inline docs/source/guide/glossary.md:63]:
```suggestion
: The error-mitigated expectation value of some observable is estimated by searching the subspace of an output quantum state for a variation of the state with the lowest error rate. This is realized without utilizing syndrome measurements often required by quantum error-correcting schemes.  
```

File: docs/source/guide/glossary.md [hunk 2]
```diff
@@ -52,6 +59,9 @@ between the qubits and the environment, mitigating the effects of noise.
 [Probabilistic Error Cancellation (PEC)](pec.md)
 : Ideal operations are represented as quasi-probability distributions over noisy implementable operations, and unbiased estimates of expectation values are obtained by averaging over circuits sampled according to this representation.
 
+[Quantum Subspace Expansion](qse.md)
+: The error-mitigated expectation value of some observable is estimated by searching the subspace of an output quantum state for a variation of the state with the lowest error rate. This is realized without utilizing syndrome measurements often required by quantum error-correcting schemes.
```
Comment (purva-thakre) [inline docs/source/guide/glossary.md:63]:
```suggestion
: The error-mitigated expectation value of some observable is estimated by searching the subspace of an output quantum state for a variation of the state with the lowest error rate. This is realized without utilizing intricate syndrome measurements often required by quantum error-correcting schemes.  
```

File: docs/source/guide/glossary.md [hunk 3]
```diff
@@ -39,8 +39,15 @@ the naive (i.e. unmitigated) method of running the same noisy input circuit $N$
 sample mean of the measurement outcomes. Also called sampling cost, it is usually reported as a multiplicative factor $C$, defined as the ratio of the QEM estimator's variance to the sample-mean estimator's variance, and meaning that
 the method needs $C \cdot N$ circuit shots to obtain the same precision as the sample-mean estimator would with only $N$ shots.
 
+[Pauli Twirling](pt.md)
```
Comment (purva-thakre) [inline docs/source/guide/glossary.md:42]:
```suggestion
[Pauli Twirling (PT)](pt.md)
```

File: docs/source/guide/glossary.md [hunk 4]
```diff
@@ -52,6 +59,9 @@ between the qubits and the environment, mitigating the effects of noise.
 [Probabilistic Error Cancellation (PEC)](pec.md)
 : Ideal operations are represented as quasi-probability distributions over noisy implementable operations, and unbiased estimates of expectation values are obtained by averaging over circuits sampled according to this representation.
 
+[Quantum Subspace Expansion](qse.md)
```
Comment (purva-thakre) [inline docs/source/guide/glossary.md:62]:
```suggestion
[Quantum Subspace Expansion (QSE)](qse.md)
```

File: docs/source/guide/glossary.md [hunk 5]
```diff
@@ -39,8 +39,15 @@ the naive (i.e. unmitigated) method of running the same noisy input circuit $N$
 sample mean of the measurement outcomes. Also called sampling cost, it is usually reported as a multiplicative factor $C$, defined as the ratio of the QEM estimator's variance to the sample-mean estimator's variance, and meaning that
 the method needs $C \cdot N$ circuit shots to obtain the same precision as the sample-mean estimator would with only $N$ shots.
 
+[Pauli Twirling (PT)](pt.md)
+: A noise-tailoring technique utilizing Pauli gates is used to tailor the noise in an input circuit to be more manageable. Coherent errors contribute heavily to the quadratically worst-case gate infidelities scenario compared to incoherent errors. This could indirectly affect the performance of a large noisy quantum circuit if the circuit noise is not tailored to be a Pauli noise channel i.e. incoherent.
```
Comment (cosenal) [inline docs/source/guide/glossary.md:43]: > A noise-tailoring technique utilizing Pauli gates is used to tailor the noise

The wording here reads a bit redundant

**Review by cosenal — APPROVED** [review]

**Comment by cosenal** [comment]
@purva-thakre Thanks for this! Feel free to merge.

**Pull request merged**